x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalLogicalPlanOptimizerTests.java

```diff
@@ -190,6 +190,65 @@ public void testMissingFieldInEval() {
         var source = as(limit.child(), EsRelation.class);
     }
 
+    /**
+     * Expects
+     * LocalRelation[[first_name{f}#4],EMPTY]
```

Comment:
If I understand this correctly, then `emp_no` is a missing field, so gets replaced with `null` ( see _ReplaceMissingFieldWithNull_ ), so becomes `null > 10`. The Filter rule then eliminates this as it can never be true.

Reply:
It's not a missing field, at least not in the mapping used by the analyzer.

I am failing to see the reason for the "emp_no" filtering being performed not on the local node. I _think_ it has to do with a potential "limit" that can be applied on the values.... if every node would return X rows sorted asc/desc, the coordinator (which has to perform the final limit and sort) can stop prematurely after receiving enough rows to fulfill the topN requirements. But, if every node returns all the rows the final result can be different (all the values can come from a single shard for example and all others are discarded).

Reply:
Ah, thanks, I missed that we use `statsForMissingField`, which acts as if `emp_no` was indeed missing from the plan.

(Nit:) In this case, maybe it would help readability a tiny bit if we maybe renamed `testStats -> sourceWithoutEmpNo` or so.

Or (out of scope for this PR) renamed `statsForMissingField -> testStatsWithoutFields` and added a javadoc string (`SearchStats` themselves would also profit from a doc string IMO).

I think I got confused about the intent of `statsForMissingField`, but that could also just come from lacking familiarity with the code base, of course ;)

Reply:
missingField in this context implies a field that exists in the mapping (otherwise the query would be invalid) but it missing locally (either there's no mapping or there are no values for it).
since emp_no doesn't exist locally it `eval x = emp_no` means `x = null` and this ends up folding the filter to false which ends up into a LocalRelation.